### PR TITLE
cleaned bad text

### PR DESCRIPTION
**Pull request by SirGryphin**
just removing bad text from chapters, with new templates it got removed I think.

### CODE REVIEW

**Comment by dipu-bd** [comment]
@SirGryphin We are currently moving away from `bad_text_regex` and should start using the [bad_tag_text_pairs](https://github.com/dipu-bd/lightnovel-crawler/blob/dev/lncrawl/utils/cleaner.py#L28).  You need to define a tag name and regex to remove them. In case we do no have the tag name, we can use `bad_text_regex`. 

Sadly, this won't work with the template. The template just uses `clean_contents`, not `extract_contents`. I have not tested it well to know if it would work with `extrac_contents`. @idMysteries Please let us know if we can use `extract_contents` safely.

**Comment by idMysteries** [comment]
@dipu-bd The text in this PR will not work because of the semicircular brackets.
Non-closed semicircular brackets in regexp will give an error.

**Comment by idMysteries** [comment]
In general, cleaner doesn't seem to be working right now. There's an error with dict I have.

**Comment by SirGryphin** [comment]
@dipu-bd @idMysteries 
I tried the following code to replace `bad_text_regex` and I get a error, I don't know if I'm doing something wrong I've tried multiple times and same thing.
```
self.cleaner.bad_tag_text_pairs.update(
            {
                'strong': "This chapter upload first at NovelNext.com",
            }
        )
```
```
? How many files to generate? Pack everything into a single file
TypeError: unhashable type: 'dict'
TypeError: unhashable type: 'dict'
TypeError: unhashable type: 'dict'
TypeError: unhashable type: 'dict'
TypeError: unhashable type: 'dict'
TypeError: unhashable type: 'dict'
TypeError: unhashable type: 'dict'
TypeError: unhashable type: 'dict'
TypeError: unhashable type: 'dict'
TypeError: unhashable type: 'dict'
Chapters: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 11.93item/s] 
  Images: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1/1 [00:00<00:00,  4.40item/s] 
```

**Comment by dipu-bd** [comment]
I will check this out

**Comment by dipu-bd** [comment]
Fixed. Try this pattern:  @SirGryphin 

```
self.cleaner.bad_tag_text_pairs.update(
    {
        'h4': r'Chapter \d+',
        'p': [
            r'^\s*(Translator|Editor):.*$',
                r"Bookmark this website",
        ],
        'strong': r'NovelNext\.com',
        'strong': r"This chapter upload first at NovelNext\.com",
    }
)
```

**Comment by SirGryphin** [comment]
Working okay, now removes `This chapter upload first at NovelNext.com` having a problem figuring out regex for remove translator and editor as there not in own p tag like normal but on one line with strong tags.

`<p><strong>Translator:</strong> Atlas Studios <strong>Editor:</strong> Atlas Studios</p>`

I  tried `^\s*(<strong>Translator|<strong>Editor):.*$` but it doesn't seem to work even though it says it should on test site below.

https://regex101.com/r/7G13C2/1

I've even tried just using `r'<strong>Translator:</strong> Atlas Studios <strong>Editor:</strong> Atlas Studios',` but it still does not remove it. Maybe I'm doing something wrong on regex101 it shows as working but does not work in python.

**Comment by dipu-bd** [comment]
> Maybe I'm doing something wrong on regex101 it shows as working but does not work in python.

Have you tried the example code I gave? I tested using this, and it worked for me. I haven't checked many samples though

**Comment by dipu-bd** [comment]
> ^\s*(<strong>Translator|<strong>Editor):.*$

regex are text only, you can not put `<strong>` there since it is a html tag.

**Comment by idMysteries** [comment]
@dipu-bd  How does Dict work for you? He has unique keys. Or do you use some special dictionary here?
image

**Comment by SirGryphin** [comment]
@dipu-bd Update bouth source files using `text_pairs` and the working fine now, the problem with removing translator and editor was me being a idiot. I was opening epub and see this:

`<p><strong>Translator:</strong> Atlas Studios <strong>Editor:</strong> Atlas Studios</p>`

I couldn't get it to be removed, figure out that it only shows like that in epub before being processed through lncrawl on main site it looks like this:

`<h4><strong>Translator: </strong>Atlas Studios <strong> Editor: </strong>Atlas Studios</h4>`

So I was trying to remove it from <p> tag instead of <h4> lol, all fixed and working. I hope I've done everything correctly this time.

File: sources/en/h/hotnovelfull.py [hunk 1]
```diff
@@ -8,3 +8,14 @@
...
     base_url = ["https://hotnovelfull.com/"]
+
+    def initialize(self) -> None:
+        self.cleaner.bad_tag_text_pairs.update(
+            {
+                'h4': [
+                    r'Chapter \d+',
+                    r'^\s*(Translator|Editor):.*$',
+                ],
+                'strong': r"This chapter upload first at NovelNext\.com",
```
Comment (dipu-bd) [inline sources/en/h/hotnovelfull.py:19]: There is NovelNext.com in HotNovelFull? What a thief 😆

**Review by dipu-bd — APPROVED** [review]

**Pull request merged**